### PR TITLE
ERC721 Enumerable

### PR DESCRIPTION
**Pull request by starknetdev**
## Introduced changes

Adds enumerable extension as a dojo component.

## Checklist

- [x] Add enumerable component
- [x] Add component level tests
- [x] Add enumerable mintable burnable preset
- [x] Add tests for the preset

### CODE REVIEW

**Comment by rsodre** [comment]
what about this one, what's missing to be merged?
A lot that I see here was already added and merged by #58, the actual diff may be a lot smaller

**Comment by starknetdev** [comment]
> what about this one, what's missing to be merged? A lot that I see here was already added and merged by #58, the actual diff may be a lot smaller @notV4l ?

rebased now 👍

File: token/src/presets/erc721/tests_enumerable_mintable_burnable.cairo [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+            erc_721_token_approval_model::TEST_CLASS_HASH,
+            erc_721_balance_model::TEST_CLASS_HASH,
+            erc_721_meta_model::TEST_CLASS_HASH,
+        ]
+    );
+
+    // deploy contract
+    let mut erc721_enum_mint_burn_dispatcher = IERC721EnumMintBurnPresetDispatcher {
+        contract_address: world
+            .deploy_contract('salt', ERC721EnumMintBurn::TEST_CLASS_HASH.try_into().unwrap())
```
Comment (notV4l) [inline token/src/presets/erc721/tests_enumerable_mintable_burnable.cairo:61]: it misses calldata

File: token/src/presets/erc721/enumerable_mintable_burnable.cairo [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    impl InitializerImpl of super::IERC721EnumInit<ContractState> {
+        fn initializer(
+            ref self: ContractState,
+            name: ByteArray,
+            symbol: ByteArray,
+            base_uri: ByteArray,
+            recipient: ContractAddress,
+            token_ids: Span<u256>
+        ) {
+            self.erc721_metadata.initialize(name, symbol, base_uri);
```
Comment (notV4l) [inline token/src/presets/erc721/enumerable_mintable_burnable.cairo:189]: we can add test on world owner & use initializable component same way than in other preset

File: token/src/lib.cairo
Comment (notV4l) [inline token/src/lib.cairo:1]: missing 
```
mod enumerable_mintable_burnable;
#[cfg(test)]
mod tests_enumerable_mintable_burnable;
```

**Pull request merged**